lntest/harness.go

```diff
@@ -171,6 +171,104 @@ func (h *HarnessTest) Context() context.Context {
+// associated remote signing instance.
+func (h *HarnessTest) setupWatchOnlyNode(name string,
+	signerNode *node.HarnessNode, password []byte) *node.HarnessNode {
+
+	// Prepare arguments for watch-only node connected to the remote signer.
+	remoteSignerArgs := []string{
+		"--remotesigner.enable",
+		fmt.Sprintf("--remotesigner.rpchost=localhost:%d",
+			signerNode.Cfg.RPCPort),
+		fmt.Sprintf("--remotesigner.tlscertpath=%s",
+			signerNode.Cfg.TLSCertPath),
+		fmt.Sprintf("--remotesigner.macaroonpath=%s",
+			signerNode.Cfg.AdminMacPath),
+	}
+
+	// Fetch watch-only accounts from the signer node.
+	resp := signerNode.RPC.ListAccounts(&walletrpc.ListAccountsRequest{})
+	watchOnlyAccounts, err := walletrpc.AccountsToWatchOnly(resp.Accounts)
+	if err != nil {
+		h.Fatalf("unable to find watch only accounts for %s: %v",
+			name, err)
+	}
+
+	// Create a new watch-only node with remote signer configuration.
+	return h.NewNodeRemoteSigner(name, remoteSignerArgs, password,
+		&lnrpc.WatchOnly{
+			MasterKeyBirthdayTimestamp: 0,
+			MasterKeyFingerprint:       nil,
+			Accounts:                   watchOnlyAccounts,
+		},
+	)
+}
+
+// createAndSendOutput send amt satoshis from the internal mining node to the
+// targeted lightning node using a P2WKH address. No blocks are mined so
+// transactions will sit unconfirmed in mempool.
+func (h *HarnessTest) createAndSendOutput(target *node.HarnessNode,
+	amt btcutil.Amount, addrType lnrpc.AddressType) {
+
+	req := &lnrpc.NewAddressRequest{Type: addrType}
+	resp := target.RPC.NewAddress(req)
+	addr := h.DecodeAddress(resp.Address)
+	addrScript := h.PayToAddrScript(addr)
+
+	output := &wire.TxOut{
+		PkScript: addrScript,
+		Value:    int64(amt),
+	}
+	h.Miner.SendOutput(output, defaultMinerFeeRate)
+}
+
+// SetupRemoteSigningStandbyNodes starts the initial seeder nodes within the
+// test harness in a remote signing configuration. The initial node's wallets
+// will be funded wallets with 100x1 BTC outputs each.
+func (h *HarnessTest) SetupRemoteSigningStandbyNodes() {
+	h.Log("Setting up standby nodes Alice and Bob with remote " +
+		"signing configurations...")
+	defer h.Log("Finished the setup, now running tests...")
+
+	password := []byte("itestpassword")
+
+	// Setup remote signing nodes for Alice and Bob.
+	signerAlice := h.NewNode("SignerAlice", nil)
+	signerBob := h.NewNode("SignerBob", nil)
+
+	// Setup watch-only nodes for Alice and Bob, each configured with their
+	// own remote signing instance.
+	h.Alice = h.setupWatchOnlyNode("Alice", signerAlice, password)
+	h.Bob = h.setupWatchOnlyNode("Bob", signerBob, password)
+
+	// Fund each node with 100 BTC (using 100 separate transactions).
+	const fundAmount = 1 * btcutil.SatoshiPerBitcoin
+	const numOutputs = 100
+	const totalAmount = fundAmount * numOutputs
+	for _, node := range []*node.HarnessNode{h.Alice, h.Bob} {
+		h.manager.standbyNodes[node.Cfg.NodeID] = node
+		for i := 0; i < numOutputs; i++ {
+			h.createAndSendOutput(node, fundAmount,
```

Comment:
I tried using `h.FundCoins(...)`, but found that to considerably lengthen the setup time of the standby nodes. I think probably because we opt to create 100 outputs per node and it's mining and waiting for sync after each output created. That led me to [`h.FundCoinsUnconfirmed(...)`](https://github.com/lightningnetwork/lnd/blob/v0.17.3-beta/lntest/harness.go#L1336) where I had trouble with the [assertion of # of UTXO](https://github.com/lightningnetwork/lnd/blob/v0.17.3-beta/lntest/harness.go#L1296) in mempool on repeated invocation. If the test is agnostic to funding each node with 1 large output vs. 100 smaller outputs, then I think I could use the `h.FundCoinsUnconfirmed(...)` as is.

Reply:
Nice to have this method extracted out! Yeah, we don't want to use `FundCoins` as it mines and verifies after each send as you describe. So this looks good!

Just a nit: formatting of multi-line function call should be:
```
			h.createAndSendOutput(
				node, fundAmount,
				lnrpc.AddressType_WITNESS_PUBKEY_HASH,
			)
```

Reply:
Updated! ✅